### PR TITLE
LPF-190 - get tests to talk to payforlegalaid api

### PR DESCRIPTION
**Pull request by ChrisCookOC**
Update pipeline name to be more sensible
Add test that talks to payforlegalaid api

The CucumberHooks file contains a BeforeAll and AfterAll hook that before the tests starts up the service (running it using java -jar) and afterwards shuts it down. As its in BeforeAll it needs to be static and so the ServiceManager is static, but if we need to in future we can change that.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -56,6 +56,21 @@
             <artifactId>junit-jupiter</artifactId>
             <scope>test</scope>
         </dependency>
+
+        <dependency>
+            <groupId>org.apache.httpcomponents</groupId>
+            <artifactId>httpclient</artifactId>
+            <version>4.5.14</version>
```
Comment (RobertBuczek) [inline pom.xml:63]: **Suggestion**: Would be recommended to move the hardcoded dependency versions (e.g., httpclient version 4.5.14) to a <properties> block.
**Reason**:
This centralises version management, making it easier to update dependencies across the project in the future. Here's an example:
```
<properties>
    <httpclient.version>4.5.14</httpclient.version>
</properties>
```

File: src/test/java/uk/gov/laa/pfla/ServiceManager.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+        String jarPath = new File("target/libs/pay-for-legal-aid-0.0.1-SNAPSHOT.jar").getAbsolutePath();
+        ProcessBuilder processBuilder = new ProcessBuilder("java", "-jar", "-Dspring.profiles.active=local", jarPath);
+        processBuilder.redirectOutput(ProcessBuilder.Redirect.INHERIT);
+        processBuilder.redirectError(ProcessBuilder.Redirect.INHERIT);
+        serviceProcess = processBuilder.start();
+    }
+
+    public static void stopService() {
+        if (serviceProcess != null && serviceProcess.isAlive()) {
+            serviceProcess.destroy();
```
Comment (RobertBuczek) [inline src/test/java/uk/gov/laa/pfla/ServiceManager.java:19]: **Suggestion**: Add a more robust shutdown mechanism for the service by using destroyForcibly() https://docs.oracle.com/javase/8/docs/api/java/lang/Process.html#destroyForcibly and a shutdown hook.
**Reason**:
Using destroyForcibly() ensures that the service is properly terminated, even if it becomes unresponsive. Additionally, a shutdown hook ensures that resources are released even if the JVM exits unexpectedly:

File: src/test/java/uk/gov/laa/pfla/ServiceManager.java [hunk 3]
```diff
@@ -0,0 +1,23 @@
+package uk.gov.laa.pfla;
+
+import java.io.File;
+import java.io.IOException;
+
+public class ServiceManager {
```
Comment (RobertBuczek) [inline src/test/java/uk/gov/laa/pfla/ServiceManager.java:6]: **Suggestion**: Add logging when starting and stopping the service.
**Reason**:
This provides visibility into the lifecycle of the service, which can be invaluable for debugging if the service fails to start or stop correctly.

File: src/test/java/uk/gov/laa/pfla/StepDefinitions.java [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+                        response = httpClient.execute(request);
+                    }
+                }
+        );
+
+    }
+
+    @Then("it should return a 200 response")
+    public void it_should_return_200() {
+        assertEquals(200, response.getStatusLine().getStatusCode());
```
Comment (RobertBuczek) [inline src/test/java/uk/gov/laa/pfla/StepDefinitions.java:39]: **Suggestion**: Provide detailed messages in assertions, such as when verifying the HTTP response status.
**Reason**:
Detailed assertion messages make it easier to understand why a test failed, especially when examining CI logs:
```
assertEquals(Response.SC_OK, response.getStatusLine().getStatusCode(), 
    "Expected HTTP 200 but received " + response.getStatusLine());
```

File: src/test/resources/uk/gov/laa/gpfd/health_api.feature [hunk 5]
```diff
@@ -0,0 +1,11 @@
+Feature: Call Actuator Endpoint
+
+  Scenario: Return 200 response
+    Given the service is running and we are not logged in
+    When it calls the actuator endpoint
+    Then it should return a 200 response
+
+  Scenario: Return 201 response
```
Comment (RobertBuczek) [inline src/test/resources/uk/gov/laa/gpfd/health_api.feature:8]: **Suggestion**: Double-check the scenario. It appears to expect a 201 response but verifies against 200.

Comment (ChrisCookOC) [reply src/test/resources/uk/gov/laa/gpfd/health_api.feature:8]: That was a second test to check the BeforeAll stuff was working properly, should have been deleted 🙈 Nice spot

**Review by H3ll3m4 — APPROVED** [review]
🚀

**Pull request merged**